spec/system/distribution_system_spec.rb

```diff
@@ -37,7 +39,9 @@
 
         select @partner.name, from: "Partner"
         expect do
-          click_button "Save"
+          accept_alert do
+            click_button "Save"
+          end
```

Comment:
Add a `page.find('.alert')` here to make Capybara wait for the alert before performing the `not_to change { ActionMailer::Base.deliveries.count }`

Reply:
After the `accept_alert` block still inside the `expect` block.

Reply:
Hello @sobrinho -- I was wondering what the `page.find('.alert')` would prevent? Would Capybara not wait for the `alert` using `accept_alert` by itself. Geniunely curious, I haven't used Capybara extensively.

Reply:
Hello! It will prevent the `not_to change { ActionMailer::Base.deliveries.count }` to be executed before the alert appears on the page.

If the application takes like 3 seconds to show the alert and there is a bug that delivers the message, the assertion may be executed before the message is really delivered.

You can emulate this problem by implementing the bug, adding a delay of 3 seconds before showing the message, and running the spec as-is and after with this "wait trick".

Reply:
@sobrinho I like the way you're thinking here!

In terms of ActionMailer::Base.deliveries, everything needed to add new emails to that I think is done synchronously because we're in test mode, and so completing the request will make sure that all finishes.

Also, anything that's done asynchronously would recur regardless of whether we add this page.find here, since we don't block user action while waiting to process emails, anyways.

Reply:
But here is the thing, clicking in the button won't wait for the page to finish loading, it is "async", that's why it might happen.

Try adding a `sleep 5` in the controller and send the email (by mistake for testing purposes) and run this spec again, it won't fail ;)